cmd/bosun/sched/template.go

```diff
@@ -230,78 +232,96 @@ func (c *Context) eval(v interface{}, filter bool, series bool, autods int) (res
 	default:
 		return nil, "", fmt.Errorf("expected string, expression or resultslice, got %T (%v)", v, v)
 	}
-	if filter {
+	if filter && err != nil {
 		res = res.Filter(c.AlertKey.Group())
 	}
-	if series {
+	if series && err != nil {
 		for _, k := range res {
 			if k.Type() != models.TypeSeriesSet {
 				return nil, "", fmt.Errorf("need a series, got %v (%v)", k.Type(), k)
 			}
 		}
 	}
-	return
+	return res, title, err
 }
 
 // Lookup returns the value for a key in the lookup table for the context's tagset.
-func (c *Context) Lookup(table, key string) (string, error) {
+// the returned string may be the representation of an error
+func (c *Context) Lookup(table, key string) string {
 	return c.LookupAll(table, key, c.AlertKey.Group())
 }
 
-func (c *Context) LookupAll(table, key string, group interface{}) (string, error) {
+func (c *Context) LookupAll(table, key string, group interface{}) string {
 	var t opentsdb.TagSet
 	switch v := group.(type) {
 	case string:
 		var err error
 		t, err = opentsdb.ParseTags(v)
 		if err != nil {
-			return "", err
+			c.addError(err)
+			return err.Error()
 		}
 	case opentsdb.TagSet:
 		t = v
 	}
 	l := c.schedule.RuleConf.GetLookup(table)
 	if l == nil {
-		return "", fmt.Errorf("unknown lookup table %v", table)
+		err := fmt.Errorf("unknown lookup table %v", table)
+		c.addError(err)
+		return err.Error()
```

Comment:
Unless there are cases where "" is a valid result, not sure if there are any, but not fond of the precedent. I think I would rather see errors in the templates for items that lose the entire template in case of strings and images.